src/Platform/Win/WinPlatform.cpp

```diff
@@ -9,66 +9,78 @@
-        current_key[delimiter - path] = '\0';
+        wcsncpy(current_key, path, delimiter - path);
+        current_key[delimiter - path] = L'\0';
 
-        strcpy(path_tail, delimiter + 1);
+        wcscpy(path_tail, delimiter + 1);
 
         if (!parent_key) {
-            if (iequals(current_key, "HKEY_CLASSES_ROOT"))
+            if (!wcsicmp(current_key, L"HKEY_CLASSES_ROOT"))
                 parent_key = HKEY_CLASSES_ROOT;
-            else if (iequals(current_key, "HKEY_CURRENT_CONFIG"))
+            else if (!wcsicmp(current_key, L"HKEY_CURRENT_CONFIG"))
                 parent_key = HKEY_CURRENT_CONFIG;
-            else if (iequals(current_key, "HKEY_CURRENT_USER"))
+            else if (!wcsicmp(current_key, L"HKEY_CURRENT_USER"))
                 parent_key = HKEY_CURRENT_USER;
-            else if (iequals(current_key, "HKEY_LOCAL_MACHINE"))
+            else if (!wcsicmp(current_key, L"HKEY_LOCAL_MACHINE"))
                 parent_key = HKEY_LOCAL_MACHINE;
-            else if (iequals(current_key, "HKEY_USERS"))
+            else if (!wcsicmp(current_key, L"HKEY_USERS"))
                 parent_key = HKEY_USERS;
             else
                 return false;
 
-            delimiter = strstr(path_tail, "/");
+            delimiter = wcsstr(path_tail, L"/");
             if (delimiter) {
-                strncpy(current_key, path_tail, delimiter - path_tail);
-                current_key[delimiter - path_tail] = '\0';
+                wcsncpy(current_key, path_tail, delimiter - path_tail);
+                current_key[delimiter - path_tail] = L'\0';
 
-                strcpy(path_tail, delimiter + 1);
+                wcscpy(path_tail, delimiter + 1);
             } else {
                 return false;
             }
         }
 
-        if (!strcmp(current_key, "WOW6432Node")) {
+        if (!wcscmp(current_key, L"WOW6432Node")) {
             return OS_GetAppStringRecursive(parent_key, path_tail, out_string, out_string_size, KEY_WOW64_32KEY);
         }
 
         bool result = false;
         HKEY key;
-        if (!RegOpenKeyExA(parent_key, current_key, 0, KEY_READ | flags, &key)) {
+        if (!RegOpenKeyExW(parent_key, current_key, 0, KEY_READ | flags, &key)) {
             result = OS_GetAppStringRecursive(key, path_tail, out_string, out_string_size, 0);
             RegCloseKey(key);
         }
 
         return result;
     } else {
-        DWORD data_size = out_string_size;
-        if (RegQueryValueExA(parent_key, path, nullptr, nullptr, (LPBYTE)out_string, &data_size))
-            return false;
-        return true;
+        wchar_t *tmp = new wchar_t[out_string_size + 1];
+        DWORD data_size = sizeof(wchar_t) * (out_string_size + 1);
+        LSTATUS status = RegQueryValueExW(parent_key, path, nullptr, nullptr, (LPBYTE)tmp, &data_size);
+        if (status == ERROR_SUCCESS) {
+            int numBytesConverted = WideCharToMultiByte(CP_ACP, 0, tmp, -1, out_string, out_string_size, nullptr, nullptr);
```

Comment:
Please use CP_UTF8

Reply:
GP_ACP is the window's standard. and windows does not work with utf8. why would I want to feed windows with utf8 strings? that's exactly the problem i'm solving in the first place - due to our source files having utf8 encoding, c-string literal included utf8 special symbols and was breaking some following winapi calls. i want precisely CP_ACP here onwards

however i've been thinking about that afterwards and there's a valid use case where we don't handle unicode paths (at least on windows in the current impl)
so it makes sense to return a utf8 path here only if later on we convert it to wchar_t before feeding to winapi, somewhere in the platform code
i can see that this path is later fed into std::filesystem - does it operate in utf8 on windows, or is it ascii?

Reply:
std::filesystem uses the system-specific encoding.

But this discussion opens a can of worms that's way too big.

OK let's leave it as CP_ACP and leave a TODO to switch to UTF8 once we have a solution for the bigger problem.